### PR TITLE
Use dependency injection and fallback on container

### PR DESCRIPTION
**Pull request by greg0ire**
## Subject
I am targeting this branch, because this is BC.

## Changelog

```markdown
### Fixed
- exception when calling `sonata:user:two-step-verification` even with proper configuration

### Deprecated
- executing a `Sonata\UserBundle\Command\TwoStepVerificationCommand` that did not receive arguments.
```

# How can I test this?

```shell
composer config repositories.greg0ire vcs https://github.com/greg0ire/SonataUserBundle
composer require sonata-project/user-bundle "dev-fix-two-step-verif-command as 4.2.3"
```

### CODE REVIEW

File: src/Command/TwoStepVerificationCommand.php [hunk 1]
```diff
@@ -21,6 +23,26 @@
...
+
+    /**
+     * @var ?UserManagerInterface
+     */
+    private $userManager;
+
+    public function __construct(
+        ?string $name,
+        ?Helper $helper = null,
+        UserManagerInterface $userManager = null
```
Comment (OskarStark) [inline src/Command/TwoStepVerificationCommand.php]: why not `?UserManagerInterface` like `?Helper` ?

File: src/Command/TwoStepVerificationCommand.php [hunk 2]
```diff
@@ -21,6 +23,26 @@
...
+    ) {
+        parent::__construct($name);
```
Comment (OskarStark) [inline src/Command/TwoStepVerificationCommand.php]: I would prefer a blank line afterwards for readability

File: src/Command/TwoStepVerificationCommand.php (hunk 1 above)
Comment (greg0ire) [reply src/Command/TwoStepVerificationCommand.php]: Sorry, that is a mistake

**Comment by greg0ire** [comment]
This was reported by someone on Slack, and we should wait for him to test it before merging this.

**Comment by OskarStark** [comment]
Please fix FlintCI

**Comment by Jean-ita** [comment]
Hi,
i am the one who reported the issue on the slack channel.
I just tested this PR and all works as it should.
With the proper config:
`sonata_user:
    google_authenticator:
        enabled: true
        server:  yourserver.com`
It works

With no congif it correctly throws the error "Two Step Verification process is not enabled"
with this stack trace
` () at /var/www/symfony4/vendor/sonata-project/user-bundle/src/Command/TwoStepVerificationCommand.php:77
 Sonata\UserBundle\Command\TwoStepVerificationCommand->execute() at /var/www/symfony4/vendor/symfony/console/Command/Command.php:255
 Symfony\Component\Console\Command\Command->run() at /var/www/symfony4/vendor/symfony/console/Application.php:919
 Symfony\Component\Console\Application->doRunCommand() at /var/www/symfony4/vendor/symfony/framework-bundle/Console/Application.php:89
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at /var/www/symfony4/vendor/symfony/console/Application.php:262
 Symfony\Component\Console\Application->doRun() at /var/www/symfony4/vendor/symfony/framework-bundle/Console/Application.php:75
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /var/www/symfony4/vendor/symfony/console/Application.php:145
 Symfony\Component\Console\Application->run() at /var/www/symfony4/bin/console:38`

**Pull request merged**